Frameworks/UIKit/UITextField.mm

```diff
@@ -116,13 +122,21 @@ @implementation UITextField {
 - (void)setText:(NSString*)text {
     [_secureModeLock lock];
     if (_secureTextMode) {
-        _passwordBox.password = [text copy];
+        _passwordBox.Password(objcwinrt::string(text));
     } else {
-        _textBox.text = [text copy];
-        // Ensure caret at end of field in case we programmatically
-        // gain focus (becomeFirstResponder) after the text is set:
-        _textBox.selectionStart = [text length];
-        _textBox.selectionLength = 0;
+        _textBox.Text(objcwinrt::string(text));
+
+        try {
+            // Ensure caret at end of field in case we programmatically
+            // gain focus (becomeFirstResponder) after the text is set:
+            _textBox.SelectionStart([text length]);
+            _textBox.SelectionLength(0);
+        } catch (const winrt::hresult_error& e) {
+            // Bug in XAML TextBox causes it to return S_FALSE
```

Comment:
>Bug in XAML TextBox [](start = 15, length = 19)

Newer versions of the C++/WinRT headers have been changed to swallow these exceptions. Once we pick up those headers, we can remove this try/catch block.